cfme/infrastructure/virtual_machines.py

```diff
@@ -391,19 +391,21 @@ def active(self):
             except CandidateNotFound:
                 return False
 
+        def _create_snapshot(self, is_vm_running, has_name):
+            dct = {}
+            dct.update({'description': self.description})
+            if is_vm_running:
+                dct.update({'snapshot_memory': self.memory})
+            if has_name is not None:
+                dct.update({'name': self.name})
```

Comment:
all those single element updates can use the item update syntax

the arguments for triggering actions do not match their actions

**REQUIRED** the name argument should either be an optional  name, or a flag that clearly communicates that it copies the own name
**REQUIRED** the argument for trapshooting the own memory should convey that in its name,
is_vm_running is simply completely removed from the effect 

in case the arguments are really to be implied, that should happen to avoid mistakes